### ISSUE TITLE
bugs related to gestures and Exit on click

### ISSUE THREAD

**Issue by shadowelite-sec** [KZDKM/Hyprspace#18]
I found a few bugs. I'm thinking of dropping them all into one issue instead of separate ones. I don't know if it's good or not :face_in_clouds: 

1. When the mouse cursor is on overview tabs, the 3-finger gesture is not working properly, glitching

https://github.com/KZDKM/Hyprspace/assets/54718375/f72efcc7-4438-4647-8cc2-c95f685cdafb

2. When I use "Exit on click" on an opened program window, the window behaves oddly, as shown in the video. Additionally, upon exiting the overview, it would be nice to have a smooth animation for that transition. 

https://github.com/KZDKM/Hyprspace/assets/54718375/c55ff9ff-4538-4b19-9aa2-fd4a91b56dd0

3. When I use a gesture to switch workspace windows, the windows in that workspace resize (adjust) only after I remove my finger. Because of this, the animation looks really bad. It would be nice to have an option to resize windows when switching workspaces(like using keybind for switching workspaces), even if I don't remove my finger.

https://github.com/KZDKM/Hyprspace/assets/54718375/697177b6-a4de-4a65-a53d-9b906d5e2be8

**Comment by KZDKM** [comment]
1 is working as intended, when you have too many workspaces you should be able to scroll through the panel, should be improved by disabling scroll when the panel is not overflown. Also, I dont know why but when I try to use two-finger scroll for the panel it does not work.

2 is mainly a easy to fix quality of life issue which is known but I did not bother to fix after I finished implementing input.

3 is a bit trickier but should be fixable.

Good suggestions!

**Comment by KZDKM** [comment]
> Another related issue is that when the overview is activated using the 3 finger swipe down gesture, but is exited by clicking on a window, the next time a gesture is used to switch to another workspace it also causes the overview to show up. The behavior is only reset by again activating and deactivating the overview using the gesture

Thanks for reporting, this is fixed in the most recent commit.

**Comment by maydayv7** [comment]
Thanks a lot!

**Comment by shadowelite-sec** [comment]
> I found a few bugs. I'm thinking of dropping them all into one issue instead of separate ones. I don't know if it's good or not 😶‍🌫️
> 
>     1. When the mouse cursor is on overview tabs, the 3-finger gesture is not working properly, glitching
> 
> 
> 2024-04-14.14-25-04.mp4
> 
>     2. When I use "Exit on click" on an opened program window, the window behaves oddly, as shown in the video. Additionally, upon exiting the overview, it would be nice to have a smooth animation for that transition.
> 
> 
> 2024-04-14.14-29-09.mp4
> 
>     3. When I use a gesture to switch workspace windows, the windows in that workspace resize (adjust) only after I remove my finger. Because of this, the animation looks really bad. It would be nice to have an option to resize windows when switching workspaces(like using keybind for switching workspaces), even if I don't remove my finger.
> 
> 
> 2024-04-14.14-29-55.mp4

@KZDKM any updates ?

**Comment by bertin0** [comment]
for 3. do you think it would be possible to scale down the whole workspace (like on gnome) instead of resizing the windows? (without forcing the apps to redraw/resize)

**Comment by KZDKM** [comment]
> for 3. do you think it would be possible to scale down the whole workspace (like on gnome) instead of resizing the windows? (without forcing the apps to redraw/resize)

Yes, that is possible. I thought of implementing it last year when I was first developing this plugin. However, it kinda overcomplicates the entire render process and is especially hard to maintain under the new render pipeline and constantly changing upstream.

**Comment by bertin0** [comment]
> Yes, that is possible. I thought of implementing it last year when I was first developing this plugin. However, it kinda overcomplicates the entire render process and is especially hard to maintain under the new render pipeline and constantly changing upstream.

Would it be easier to at least have the window sizes updated on each swipe update event?
